pkg/util/provider/machinecontroller/machine_util.go

```diff
@@ -1106,21 +1109,27 @@ func (c *controller) drainNode(ctx context.Context, deleteMachineRequest *driver
 				c.nodeLister,
 				c.volumeAttachmentHandler,
 			)
+			klog.V(3).Infof("(drainNode) Invoking RunDrain, forceDeleteMachine: %t, forceDeletePods: %t, timeOutDuration: %s", forceDeletePods, forceDeleteMachine, timeOutDuration)
 			err = drainOptions.RunDrain(ctx)
 			if err == nil {
 				// Drain successful
 				klog.V(2).Infof("Drain successful for machine %q ,providerID %q, backing node %q. \nBuf:%v \nErrBuf:%v", machine.Name, getProviderID(machine), getNodeName(machine), buf, errBuf)
 
-				description = fmt.Sprintf("Drain successful. %s", machineutils.InitiateVMDeletion)
+				if forceDeletePods {
+					description = fmt.Sprintf("Drain successful. %s", machineutils.DelVolumesAttachments)
+					err = fmt.Errorf(description)
+				} else { // regular drain already waits for vol detach and attach for another node.
+					description = fmt.Sprintf("Drain successful. %s", machineutils.InitiateVMDeletion)
+					err = fmt.Errorf("Machine deletion in process. " + description)
```

Comment:
lets delete `L1120` and move this err initialization out of else block , this can stay common for both force and graceful drain situations

Reply:
Used common `err` init for both branches to simply after the branches

```go
err = fmt.Errorf(description)
```